fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -563,11 +569,31 @@ public void replaceProperties(final IdentifierConverter<Resource, FedoraResource
             exceptions.append(e.getMessage());
         }
 
+        removeEmptyFragments();
+
         if (exceptions.length() > 0) {
             throw new MalformedRdfException(exceptions.toString());
         }
     }
 
+    private void removeEmptyFragments() {
+        try {
+            if (node.hasNode("#")) {
```

Comment:
Is there any need for this test when the loop below won't execute if there are no # nodes?